### ISSUE TITLE
Non deterministic behaviour linting multiple bbappends for the same package

### ISSUE THREAD

**Issue by robertsemaskevic** [priv-kweihmann/oelint-adv#205]
The issue described below looks similar to https://github.com/priv-kweihmann/oelint-adv/issues/203.

Linter randomly reports false positives for the task order if multiple bbappends for the same package are linted in one go. However, linting the same files in isolation doesn't trigger any warnings.

Linting multiple files:
```
[08:10:57] $ oelint-adv meta-foobar/recipes-foobar/foobar.bbappend meta-foobar2/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"

[08:10:58] $ oelint-adv meta-foobar/recipes-foobar/foobar.bbappend meta-foobar2/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"
.../meta-foobar2/recipes-foobar/foobar.bbappend:1:warning:oelint.task.order.do_compile:'do_compile' should be placed before 'do_install'
[08:10:59] $ oelint-adv meta-foobar/recipes-foobar/foobar.bbappend meta-foobar2/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"
.../meta-foobar2/recipes-foobar/foobar.bbappend:1:warning:oelint.task.order.do_compile:'do_compile' should be placed before 'do_install'
[08:11:00] $ oelint-adv meta-foobar/recipes-foobar/foobar.bbappend meta-foobar2/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"
.../meta-foobar2/recipes-foobar/foobar.bbappend:1:warning:oelint.task.order.do_compile:'do_compile' should be placed before 'do_install'
[08:11:01] $ oelint-adv meta-foobar/recipes-foobar/foobar.bbappend meta-foobar2/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"

[08:11:02] $
```

Each file in isolation:
```
[08:11:02] $ oelint-adv meta-foobar/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"

[08:11:18] $ oelint-adv meta-foobar/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"

[08:11:19] $ oelint-adv meta-foobar/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"

[08:11:20] $ oelint-adv meta-foobar2/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"

[08:11:36] $ oelint-adv meta-foobar2/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"

[08:11:37] $ oelint-adv meta-foobar2/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"

[08:11:37] $ oelint-adv meta-foobar2/recipes-foobar/foobar.bbappend --quiet --color --suppress="oelint.var.bbclassextend"

[08:11:38] $
```

The above happens if bbappends containing only a single install or compile task are in different layers:
```
├── meta-foobar
│   ├── conf
│   │   └── layer.conf
│   └── recipes-foobar
│       └── foobar.bbappend
└── meta-foobar2
    ├── conf
    │   └── layer.conf
    └── recipes-foobar
        └── foobar.bbappend

[08:25:06] $ cat meta-foobar/recipes-foobar/foobar.bbappend
do_install_prepend() {
    echo "foobar install"
}
[08:25:16] $ cat meta-foobar2/recipes-foobar/foobar.bbappend
do_compile_prepend() {
    echo "foobar compile"
}
```

Reproduced with oelint-parser 1.0.10, oelint-adv 2.2.2, python 3.6|3.9.

**Comment by priv-kweihmann** [comment]
I will have a look, thanks for reporting
